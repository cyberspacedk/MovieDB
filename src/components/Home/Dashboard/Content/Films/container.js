import { withRouter } from 'react-router-dom';
import Films from './component';

export default withRouter(Films);
