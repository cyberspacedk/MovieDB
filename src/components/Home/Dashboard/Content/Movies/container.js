import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Movies from './component';
import {
  isError,
  isLoading,
  isEmpty,
  getSearchResponse,
  getTotalResults,
  getQuery,
  getCurrentPage,
} from '../../../../../store/search/selectors';
import { searchRequest } from '../../../../../store/search/actions';

const MoviesContainer = props => {
  const goToNextPage = page => {
    props.searchRequest(props.query, page);
  };
  return <Movies {...props} goToNextPage={goToNextPage} />;
};

MoviesContainer.propTypes = {
  query: PropTypes.string.isRequired,
  searchRequest: PropTypes.func.isRequired,
};

const mstp = state => ({
  loading: isLoading(state),
  error: isError(state),
  empty: isEmpty(state),
  movies: getSearchResponse(state),
  query: getQuery(state),
  totalResults: getTotalResults(state),
  currentPage: getCurrentPage(state),
});

const mdtp = {
  searchRequest,
};

export default compose(
  connect(
    mstp,
    mdtp,
  ),
  withRouter,
)(MoviesContainer);
