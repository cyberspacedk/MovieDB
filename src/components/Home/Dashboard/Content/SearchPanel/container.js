import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import SearchPanel from './component';
import { searchRequest } from '../../../../../store/search/actions';

export const mapPropsToValues = () => ({
  search: '',
});

export const handleSubmit = (values, { props }) => {
  const { search } = values;
  const { searchRequest } = props;
  searchRequest(search);
};

export const validationSchema = Yup.object().shape({
  search: Yup.string()
    .min(3, 'Минимум 3 символа')
    .matches(/^[A-Za-z0-9_-\s]{3,16}$/, 'Только буквы, цифры и тире')
    .required(),
});

const mdtp = {
  searchRequest,
};

export default compose(
  connect(
    null,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(SearchPanel);
