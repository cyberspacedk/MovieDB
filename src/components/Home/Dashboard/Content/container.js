import { connect } from 'react-redux';
import Content from './component';
import { singleFilmRequest } from '../../../../store/singleFilm/actions';
import {
  isError,
  isLoading,
  searchResponse,
  isEmptyResponse,
  totalResults,
} from '../../../../store/search/selectors';

export const mstp = state => ({
  isLoading: isLoading(state),
  isError: isError(state),
  searchResponse: searchResponse(state),
  isEmpty: isEmptyResponse(state),
  totalResults: totalResults(state),
});

const mdtp = {
  singleFilmRequest,
};
export default connect(
  mstp,
  mdtp,
)(Content);
