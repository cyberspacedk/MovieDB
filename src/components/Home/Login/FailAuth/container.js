import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { tryAgain } from '../../../../store/authentifiction/actions';
import FailAuth from './component';

export class FailAuthContainer extends Component {
  componentDidMount() {
    const { tryAgain } = this.props;
    setTimeout(() => {
      tryAgain();
    }, 2000);
  }

  render() {
    return <FailAuth />;
  }
}

FailAuthContainer.propTypes = {
  tryAgain: PropTypes.func.isRequired,
};

const mdtp = {
  tryAgain,
};

export default connect(
  null,
  mdtp,
)(FailAuthContainer);
