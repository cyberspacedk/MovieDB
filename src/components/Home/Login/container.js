import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import LoginForm from './component';
import { authRequest } from '../../../store/authentifiction/actions';
import { isFailAuth } from '../../../store/authentifiction/selectors';

export const mapPropsToValues = () => ({
  username: '',
  password: '',
});

export const handleSubmit = (values, { props, resetForm }) => {
  // eslint-disable-next-line no-shadow
  const { authRequest } = props;
  const { rememberMe: _, ...user } = values;
  authRequest(user);
  resetForm();
};

export const validationSchema = Yup.object().shape({
  username: Yup.string()
    .matches(/^[a-z0-9_-]{4,16}$/, 'Запрещенные символы')
    .required('Обязательное поле'),
  password: Yup.string()
    .min(4, 'Минимум 4 символов')
    .required('Обязательное поле'),
});

const mstp = state => ({
  isFailAuth: isFailAuth(state),
});

const mdtp = {
  authRequest,
};

export default compose(
  connect(
    mstp,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(LoginForm);
