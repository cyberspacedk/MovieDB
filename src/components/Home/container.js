import { connect } from 'react-redux';
import { isAuthentificated } from '../../store/authentifiction/selectors';
import Home from './component';

export const mstp = state => ({
  isAuthentificated: isAuthentificated(state),
});

export default connect(mstp)(Home);
