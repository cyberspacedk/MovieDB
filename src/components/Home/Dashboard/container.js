import React from 'react';
import Header from './Header';
import SearchPanel from './Content/SearchMovies';
import Content from './Content';

const Dashboard = () => {
  return (
    <div>
      <Header />
      <SearchPanel />
      <Content />
    </div>
  );
};

export default Dashboard;
