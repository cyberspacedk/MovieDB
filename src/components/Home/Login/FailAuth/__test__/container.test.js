import React from 'react';
import { shallow } from 'enzyme';
import { FailAuthContainer } from '../container';

describe('Component: Home', () => {
  const props = {
    tryAgain: jest.fn(),
  };
  const container = shallow(<FailAuthContainer {...props} />);

  it('Match its snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
