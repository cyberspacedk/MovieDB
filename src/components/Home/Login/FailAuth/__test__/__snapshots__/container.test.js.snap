// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Home Match its snapshot 1`] = `<FailAuth />`;
